### PR TITLE
Add tag validation into cloudformation module

### PR DESCRIPTION
**Pull request by sjambhule**

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/cloudformation.py [hunk 1]
```diff
@@ -596,6 +598,16 @@ def main():
 
     result = {}
 
+    # validate tags
+    if module.params.get('tags'):
+        team_tag = module.params['tags']['Team']
+        project_tag = module.params['tags']['Project']
+        validation_res = tag_validator.validate(team_tag, project_tag)
```
Comment (jmcorallo) [inline lib/ansible/modules/cloud/amazon/cloudformation.py]: Minor comment, `Name` tag is also required, but free form
https://dataxu.atlassian.net/wiki/spaces/dev/pages/540606620/AWS+Tags+2.0

Comment (sjambhule) [reply lib/ansible/modules/cloud/amazon/cloudformation.py]: Thanks @jmcorallo I was aware of the Name tag. 
@babineaum @scobrc do we also need to validate Name tag, can you pls clarify?

Comment (scobrc) [reply lib/ansible/modules/cloud/amazon/cloudformation.py]: Name tag will not be validated.

File: lib/ansible/modules/cloud/amazon/cloudformation.py [hunk 2]
```diff
@@ -596,6 +598,16 @@ def main():
...
+        if not validation_res:
+            module.fail_json(msg="Invalid Tag: tag validation failed.")
+    else:
+        module.fail_json(msg="Invalid Tag: global tags are not present.")
```
Comment (babineaum) [inline lib/ansible/modules/cloud/amazon/cloudformation.py]: do these error messages follow along with what ansible uses for error messages? My question is around the format of "Invalid Tag:" do they have a contribution document which might allow us to have our error messages look just like all the other ansible message?

**Comment by babineaum** [comment]
Make sure you delete your branches when you merge your PRs

**Pull request merged**